statistics/handle/statscache.go

```diff
@@ -39,47 +298,31 @@ func (key statsCacheKey) Hash() []byte {
 	return buf
 }
 
-// newStatsCache returns a new statsCache with capacity maxMemoryLimit.
-func newStatsCache(memoryLimit int64) *statsCache {
-	// Since newStatsCache controls the memory usage by itself, set the capacity of
+// newstatsCache returns a new statsCahce with capacity memoryLimit(initial 1G)
+func newSimpleStatsCache(memoryLimit int64) *simpleStatsCache {
+	// since newstatsCache controls the memory usage by itself, set the capacity of
 	// the underlying LRUCache to max to close its memory control
 	cache := kvcache.NewSimpleLRUCache(uint(memoryLimit), 0.1, 0)
-	c := statsCache{
+	c := simpleStatsCache{
 		cache:       cache,
 		memCapacity: memoryLimit,
-		memTracker:  memory.NewTracker(memory.LabelForStatsCache, -1),
+		memTracker:  memory.NewTracker(memory.LabelForStatsCache, memoryLimit),
 	}
 	return &c
 }
 
-// Clear clears the statsCache.
-func (sc *statsCache) Clear() {
-	// Since newStatsCache controls the memory usage by itself, set the capacity of
-	// the underlying LRUCache to max to close its memory control
-	sc.mu.Lock()
-	defer sc.mu.Unlock()
-	cache := kvcache.NewSimpleLRUCache(uint(sc.memCapacity), 0.1, 0)
-	sc.memTracker.ReplaceBytesUsed(0)
-	sc.cache = cache
-	sc.version = 0
+// SetBytesLimit sets the bytes limit for this tracker.
+func (sc *simpleStatsCache) SetBytesLimit(BytesLimit int64) {
+	sc.memTracker.SetBytesLimit(BytesLimit)
```

Comment:
set memCapacity.  sc.memTracker.BytesLimit shouldn't be modified

Reply:
in simplestatscache sc.memTracker.BytesLimit is used for limit the memory .